File: app/src/main/java/com/example/coffeeshop/data/model/Item.kt
package com.example.coffeeshop.data.model

import java.io.Serializable

data class Item(
    val _id: String? = null,
    val name: String,
    val image_url: String,
    val basePrice: Double,
    val description: String,
    val sizes: List<Size> = emptyList(),
    val tempOptions: List<TempOption> = emptyList(),
    val toppings: List<Topping> = emptyList(),
    val iceLevels: List<String> = emptyList(),
    val sugarLevels: List<String> = emptyList(),
    val isActive: Boolean = true,
    val categories: List<String> = emptyList()
) : Serializable

File: backend/models/Item.js
const mongoose = require('mongoose');

const SizeSchema = new mongoose.Schema({
  name: { type: String, required: true },
  modifier: { type: Number, required: true },
  label: { type: String, required: true }
});

const TempOptionSchema = new mongoose.Schema({
  name: { type: String, required: true },
  modifier: { type: Number, required: true },
  label: { type: String, required: true }
});

const ToppingSchema = new mongoose.Schema({
  name: { type: String, required: true },
  price: { type: Number, required: true }
});

const ItemSchema = new mongoose.Schema({
  name: { type: String, required: true },
  category: { 
    type: String, 
    required: true,
    enum: ['coffee', 'chocolate', 'other']
  },
  image_url: { type: String, required: true },
  basePrice: { type: Number, required: true },
  description: { type: String, default: '' },
  sizes: [SizeSchema],
  tempOptions: [TempOptionSchema],
  iceLevels: [String],
  sugarLevels: [String],
  toppings: [ToppingSchema],
  isActive: { type: Boolean, default: true }
}, {
  timestamps: true
});

module.exports = mongoose.model('Item', ItemSchema);

